xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-default/src/main/java/org/xwiki/notifications/internal/NotificationDisplayerComponent.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.notifications.internal;
+
+import java.lang.reflect.Type;
+import java.util.Arrays;
+import java.util.List;
+
+import javax.script.ScriptContext;
+
+import org.xwiki.component.manager.ComponentManager;
+import org.xwiki.component.wiki.WikiComponent;
+import org.xwiki.component.wiki.WikiComponentScope;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.EntityReference;
+import org.xwiki.notifications.CompositeEvent;
+import org.xwiki.notifications.NotificationDisplayer;
+import org.xwiki.notifications.NotificationException;
+import org.xwiki.rendering.block.Block;
+import org.xwiki.script.ScriptContextManager;
+import org.xwiki.template.Template;
+import org.xwiki.template.TemplateManager;
+import org.xwiki.text.StringUtils;
+
+import com.xpn.xwiki.objects.BaseObject;
+import com.xpn.xwiki.objects.BaseObjectReference;
+
+/**
+ * This class is meant to be instanciated and then registered to the Component Manager by the
+ * {@link NotificationDisplayerComponentBuilder} component every time a document containing a
+ * NotificationDisplayerClass is added, updated or deleted.
+ *
+ * @version $Id$
+ * @since 9.5RC1
+ */
+public class NotificationDisplayerComponent implements WikiComponent, NotificationDisplayer
```

Comment:
What about "WikiNotificationDisplayer" since that's a wiki based NotificationDisplayer ?